lib/Cake/Network/CakeSocket.php

```diff
@@ -129,8 +129,12 @@ public function connect() {
 			$this->disconnect();
 		}
 
+		$hasProtocol = strpos($this->config['host'], '://') !== false;
```

Comment:
Question answered :+1: 

The hopping from scheme to protocol to scheme is a bit awkward - but not worth worrying about.